GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -961,18 +953,18 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
                         : null;
                     ObjectId? newCurrentCheckout = headRef?.ObjectId ?? capturedModule.GetCurrentCheckout();
 
-                    // If the current checkout changed, don't get the currently selected rows, select the
-                    // new current checkout instead.
+                    // If the current checkout (HEAD) is changed, don't get the currently selected rows,
+                    // select the new current checkout instead.
                     if (newCurrentCheckout != CurrentCheckout)
                     {
-                        selectedObjectIds = null;
+                        currentlySelectedObjectIds = null;
```

Comment:
Maybe try to keep selection also when refreshing?

```suggestion
if (currentlySelectedObjectIds is not null && currentlySelectedObjectIds.Count == 1 && currentlySelectedObjectIds[0] == CurrentCheckout)
{
                        currentlySelectedObjectIds = List<ObjectId>{ newCurrentCheckout  };
}
```